sdk/core/Azure.Core.Amqp/src/AmqpAnnotatedMessage.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Azure.Core.Amqp
+{
+    /// <summary>
+    /// Represents an AMQP message.
+    /// <seealso href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#section-message-format" />
+    /// </summary>
+    public class AmqpAnnotatedMessage
+    {
+        /// <summary>
+        /// Initializes a new <see cref="AmqpAnnotatedMessage"/> instance by copying the passed in message.
+        /// </summary>
+        /// <param name="messageToCopy">The message to copy.</param>
+        public AmqpAnnotatedMessage(AmqpAnnotatedMessage messageToCopy)
+        {
+            Argument.AssertNotNull(messageToCopy, nameof(messageToCopy));
+
+            var data = messageToCopy.Body as AmqpDataMessageBody;
+            Body = new AmqpDataMessageBody(data!.Data);
+            _applicationProperties = new Dictionary<string, object>(messageToCopy.ApplicationProperties);
+            Properties = new AmqpMessageProperties(messageToCopy.Properties);
+            _messageAnnotations = new Dictionary<string, object>(messageToCopy.MessageAnnotations);
+            _deliveryAnnotations = new Dictionary<string, object>(messageToCopy.DeliveryAnnotations);
+            _footer = new Dictionary<string, object>(messageToCopy.Footer);
+            Header = new AmqpMessageHeader(messageToCopy.Header);
+        }
+
+        /// <summary>
+        /// Creates a new Data body <see cref="AmqpAnnotatedMessage"/>.
+        /// </summary>
+        /// <param name="dataBody">The data sections comprising the message body.
+        /// <seealso href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-data"/>
+        /// </param>
+#pragma warning disable CS8618 // Non-nullable field is uninitialized. Consider declaring as nullable.
+        public AmqpAnnotatedMessage(IEnumerable<BinaryData> dataBody)
+#pragma warning restore CS8618 // Non-nullable field is uninitialized. Consider declaring as nullable.
```

Comment:
Why not declare the fields nullable (`?`) instead of ignoring the error? You have logic to check to see if they are null anyway in the property getters.

Reply:
Because it is not really ever going to be null. We initialize it in the getter.

Reply:
It is null until someone calls the getter.

Reply:
But why would users need to care about this?

Reply:
This warning is not for users but for you.

Reply:
My bad.. I thought the warning was about the public properties needing to be nullable, not the private fields.